main.go

```diff
@@ -120,11 +120,25 @@ func main() {
 		os.Exit(1)
 	}
 	if strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) != "false" {
-		if err = (&glancev1.Glance{}).SetupWebhookWithManager(mgr); err != nil {
+		glanceDefaults := glancev1.GlanceDefaults{
```

Comment:
@abays quick question here: regardless of `ENABLE_WEBHOOKS`, we're always running on L127 `SetupWebhookWithManager`, is this the expected behaviour? Not sure I'm misunderstanding the purpose of `ENABLE_WEBHOOKS` variable.

Reply:
ack, I thought that not having `if err = (&glancev1.Glance{}).SetupWebhookWithManager(mgr); err != nil {` on L123 would have enabled the webhook anyway (because the `if strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) != "false" {` is closed on L125.

Reply:
I actually have a different version of this section that I forgot to push last Thursday.  Since we seem to be somewhat ambivalent on whether webhooks will only exist in OpenStack Operator or if they will also be enabled in the service operators, I have moved the setup on the defaults out of the webhook setup.  i.e. it looks something like this:

```
        // Acquire environmental defaults and initialize Glance and GlanceAPI defaults with them
	glanceDefaults := glancev1.GlanceDefaults{
		ContainerImageURL: os.Getenv("GLANCE_API_IMAGE_URL_DEFAULT"),
	}

	glanceAPIDefaults := glancev1.GlanceAPIDefaults{
		ContainerImageURL: os.Getenv("GLANCE_API_IMAGE_URL_DEFAULT"),
	}

	(&glancev1.Glance{}).Spec.SetupDefaults(glanceDefaults)
	(&glancev1.GlanceAPI{}).Spec.SetupDefaults(glanceAPIDefaults)

	// Setup webhooks if requested
	if strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) != "false" {
		if err = (&glancev1.Glance{}).SetupWebhookWithManager(mgr); err != nil {
	...
	etc
```

That way the defaults are initialized regardless of whether the service operator has webhooks enabled or not, and thus the OpenStack Operator could still properly call `Glance.Spec.Default()` in the "no service operator webhooks" context.  `SetupDefaults()` would just be placed somewhere in the `api` module like so:

```
// SetupDefaults - initialize Glance spec defaults for use with either internal or external webhooks
func (r *GlanceSpec) SetupDefaults(defaults GlanceDefaults) {
	glanceDefaults = defaults
}
```

Reply:
+1 on having the defaults initialized out of the webhook setup, and your snippet makes sense to me as it's enabled according to he `ENABLE_WEBHOOKS` parameter.
I'm assuming your're going to push the last version of it (unless other conversations are still ongoing)

Reply:
Yes, I'll push the changes now